src/main/java/com/yahoo/bullet/common/metrics/HTTPMetricEventPublisher.java

```diff
@@ -0,0 +1,157 @@
+/*
+ *  Copyright 2019, Yahoo Inc.
+ *  Licensed under the terms of the Apache License, Version 2.0.
+ *  See the LICENSE file associated with the project for terms.
+ */
+package com.yahoo.bullet.common.metrics;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.http.HttpHeaders;
+import org.apache.http.HttpStatus;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpUriRequest;
+import org.apache.http.client.methods.RequestBuilder;
+import org.apache.http.entity.ContentType;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.CloseableHttpClient;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+
+import static org.apache.http.entity.ContentType.APPLICATION_JSON;
+
+/**
+ * This class publishes a {@link MetricEvent} to a provided URL. It can retry publishes.
+ */
+@Slf4j
+public class HTTPMetricEventPublisher implements MetricEventPublisher {
+    private final String url;
+    private final String group;
+    private final int retries;
+    private final int retryIntervalMS;
+    private final Map<String, String> dimensions;
```

Comment:
Is the name "dimensions" appropriate outside of yamas?

Reply:
Yea it's just tags describing the thing in more detail